tests/libvmi/network.go

```diff
@@ -26,9 +26,22 @@ import (
 // WithInterface adds a Domain Device Interface.
 func WithInterface(iface kvirtv1.Interface) Option {
 	return func(vmi *kvirtv1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Devices.Interfaces = append(
-			vmi.Spec.Domain.Devices.Interfaces, iface,
-		)
+		// Replace the interface with the same binding
+		// in case there is any
+		ifaceReplaced := false
```

Comment:
How about dropping the comments and making it just read well with functions:
```
if ifaceIndex, ok := lookupInterface(vmi, iface); !ok {
  vmi.Spec...[ifaceIndex] = iface
} else {
   vmi...Interfaces = append(..., iface)
}
```

If you want to explicitly mention the overwrite feature, add it to the doctext, there it will fit nicely.
BTW, this will preserve the original order of the interfaces even though you overwirte it. This may not be trivial when adding the "With*" options. Maybe it will be more natural to remove the entry and just append at the tail.

Reply:
I am going to delete this it has a lot of problems and I am thinking on waitting for https://github.com/kubevirt/kubevirt/pull/4035 so I can use `NewCirros` with explicit Interface population there for the three VMs I am using here.

Reply:
Change removed solution was to remove Interfaces from created VMIs so we can add interfaces with `libvmi.Option`